src/build/inject-babel-helpers.ts

```diff
@@ -0,0 +1,55 @@
+import * as dom5 from 'dom5';
+import {fs} from 'mz';
+import * as parse5 from 'parse5';
+import * as path from 'path';
+import * as stream from 'stream';
+
+import {getFileContents} from './streams';
+
+import File = require('vinyl');
+
+
+/**
+ * When compiling to ES5 we need to inject Babel's helpers into a global so
+ * that they don't need to be included with each compiled file.
+ */
+export class InjectBabelHelpers extends stream.Transform {
+  entrypoint: string;
+
+  constructor(entrypoint: string) {
+    super({objectMode: true});
+    this.entrypoint = entrypoint;
+  }
+
+  async _transform(
+      file: File,
+      _encoding: string,
+      callback: (error?: any, file?: File) => void) {
+    if (file.path !== this.entrypoint) {
```

Comment:
Isn't it more correct / less error-prone to inject these helpers into all compiled files? Injecting into the entrypoint only will break statically-served and/or non-app-shell apps.